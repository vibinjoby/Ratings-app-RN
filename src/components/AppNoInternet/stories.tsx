import React from 'react'
import { storiesOf } from '@storybook/react-native'
import AppNoInternet from './component'

storiesOf('Components', module).add('AppNoInternet', () => <AppNoInternet />)
