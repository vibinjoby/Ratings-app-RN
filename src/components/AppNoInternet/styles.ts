import { Platform, StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.blackLessOpacity,
    zIndex: 999,
  },
  animatableView: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    height: Platform.OS === 'ios' ? 80 : 50,
    backgroundColor: Colors.red,
    paddingBottom: 20,
  },
  internetTxt: {
    color: Colors.white,
    fontSize: 16,
  },
})
