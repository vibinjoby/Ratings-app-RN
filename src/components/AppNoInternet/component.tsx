import React, { useEffect, useState } from 'react'
import { View, Modal, StatusBar, Text } from 'react-native'
import * as Animatable from 'react-native-animatable'

import styles from './styles'

interface AppNoInternetProps {
  isConnected: boolean
}

const AppNoInternet: React.FC<AppNoInternetProps> = ({ isConnected }: AppNoInternetProps) => {
  const [modalVisible, setModalVisible] = useState(false)
  const [animationType, setAnimationType] = useState('fadeInDownBig')

  useEffect(() => {
    if (isConnected) {
      setAnimationType('fadeOutUpBig')
      // hide the modal after 2 seconds so that the banner's animation is rendered
      setTimeout(() => {
        setModalVisible(false)
      }, 500)
    } else {
      setModalVisible(true)
      setAnimationType('fadeInDownBig')
    }
  }, [isConnected])

  if (isConnected) return <></>

  return (
    <Modal transparent={true} visible={modalVisible}>
      <StatusBar backgroundColor={modalVisible ? 'red' : 'grey'} />
      <View style={styles.container}>
        <Animatable.View animation={animationType} style={styles.animatableView}>
          <Text style={styles.internetTxt}>You are not connected to the Internet</Text>
        </Animatable.View>
      </View>
    </Modal>
  )
}

export default AppNoInternet
