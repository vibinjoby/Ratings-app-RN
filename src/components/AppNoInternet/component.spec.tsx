import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import AppNoInternet from './component'

describe('AppNoInternet', () => {
  it('should render', () => {
    const { toJSON } = render(<AppNoInternet isConnected={false} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should show modal when not connected to internet', () => {
    const { getByText } = render(<AppNoInternet isConnected={false} />)
    getByText('You are not connected to the Internet')
  })
})
