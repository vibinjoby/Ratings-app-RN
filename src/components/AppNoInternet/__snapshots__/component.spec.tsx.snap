// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNoInternet should render 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "flex": 1,
        "zIndex": 999,
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "height": 80,
          "justifyContent": "flex-end",
          "opacity": 0,
          "paddingBottom": 20,
          "transform": Array [
            Object {
              "translateY": -500,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        You are not connected to the Internet
      </Text>
    </View>
  </View>
</Modal>
`;
